src/Tools/ExternalAccess/AspNetCore/Internal/EmbeddedLanguages/AspNetCoreEmbeddedLanguageClassifier.cs

```diff
@@ -28,7 +28,15 @@ public void RegisterClassifications(EmbeddedLanguageClassificationContext contex
                 return;
 
             var classifiers = AspNetCoreClassifierExtensionProvider.GetExtensions(context.Project);
-            var aspContext = new AspNetCoreEmbeddedLanguageClassificationContext(context);
+            if (classifiers.Length == 0)
+                return;
+
+            var virtualChars = context.VirtualCharService.TryConvertToVirtualChars(context.SyntaxToken);
+            if (virtualChars.IsDefaultOrEmpty)
+                return;
+
+            var aspContext = new AspNetCoreEmbeddedLanguageClassificationContext(
+                context, new AspNetCoreVirtualCharSequence(virtualChars));
```

Comment:
@JamesNK I'm not actually exposing the service to you.  Instead, i just check up front if we can get the sequence of virtual-chars.  If so, then we package taht up and provide it to you in the context object for that token.